### PR TITLE
DIV-6637: Adding new test cases

### PR DESCRIPTION
**Pull request by madjava**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/DIV-6636

### Change description ###
New definitions are added in DIV-6636 and DIV-6637. Adding test cases to capture those.

https://tools.hmcts.net/jira/browse/DIV-6637

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: definitions/divorce/json/CaseEventToFields/CaseEventToFields-share-a-case-nonprod.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
     "CaseEventID": "solicitorCreate",
     "CaseFieldID": "PetitionerOrganisationPolicy",
     "PageFieldDisplayOrder": 9,
-    "DisplayContext": "OPTIONAL",
+    "DisplayContext": "COMPLEX",
```
Comment (madjava) [inline definitions/divorce/json/CaseEventToFields/CaseEventToFields-share-a-case-nonprod.json:8]: Update: Notices FR used same. @Dizrupto please advise

File: test/unit/definitions/divorce/AuthorisationComplexCaseType.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+    it('should have a list of ComplexType IDs that exist in CaseFields', () => {
+      const uniqProdComplexTypeCaseFieldID = getUniqValues(prodAuthorisationComplexType, 'CaseFieldID');
+      const uniqProdCaseFieldID = getUniqValues(allFieldsForProd, 'ID');
+
+      expect(uniqProdCaseFieldID).to.include.members(uniqProdComplexTypeCaseFieldID);
+    });
+
+  });
+});
```
Comment (madjava) [inline test/unit/definitions/divorce/AuthorisationComplexCaseType.test.js:54]: This test just verifies that the  the complex id's have been defined in the corresponding casefields definitions

File: test/unit/definitions/divorce/CaseEventToComplexTypes.test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          'CaseEventToComplexTypes'
+        ]);
+
+    });
+
+    it('should have only valid definitions', () => {
+      prodOnlyDefinitions.forEach(assertEventToComplexTypesDefinitionIsValid);
+    });
+  });
+});
```
Comment (madjava) [inline test/unit/definitions/divorce/CaseEventToComplexTypes.test.js:56]: This test validates the definition file

**Review by yarpo — APPROVED** [review]

**Review by jacmurphy — APPROVED** [review]

File: definitions/divorce/json/CaseEventToComplexTypes/CaseEventToComplexTypes-share-a-case-nonprod.json [hunk 4]
```diff
@@ -1,32 +1,35 @@
...
     "ID": "OrganisationPolicy",
     "CaseEventID": "solicitorCreate",
     "CaseFieldID": "PetitionerOrganisationPolicy",
-    "ListElementCode": "Organisation.OrganisationID",
-    "EventElementLabel": "",
-    "FieldDisplayOrder": "",
-    "DisplayContext": "MANDATORY"
+    "ListElementCode": "OrgPolicyCaseAssignedRole",
+    "DefaultValue": "[PETSOLICITOR]",
+    "FieldShowCondition": "divorceCaseNumber=\"NeverShow\"",
```
Comment (Dizrupto) [inline definitions/divorce/json/CaseEventToComplexTypes/CaseEventToComplexTypes-share-a-case-nonprod.json:21]: Need to use an existing field that is in the "solicitorCreate" event.
For example: D8PetitionerNameChangedHow

Comment (madjava) [reply definitions/divorce/json/CaseEventToComplexTypes/CaseEventToComplexTypes-share-a-case-nonprod.json]: Updated

**Pull request merged**